reference-implementation/lib/readable-stream-abstract-ops.js

```diff
@@ -118,19 +182,11 @@ export function CreateReadableStreamEnqueueFunction(stream) {
 
 export function CreateReadableStreamErrorFunction(stream) {
   return e => {
-    if (stream._state === 'waiting') {
-      stream._resolveReadyPromise(undefined);
-    }
     if (stream._state === 'readable') {
       stream._queue = [];
```

Comment:
Good point. Moved everything to ErrorReadableStream.

BTW, I'm trying to have some border between the code that is specific to the ReadableStream subclass implementation and one that is common to ReadableStream and all of its subclasses. Do you have any ideas about that?